nixos/tests/mitmproxy.nix

```diff
@@ -0,0 +1,129 @@
+      security.pki.certificateFiles = [ caCert ];
+
+      services.getty.autologinUser = "root";
+
+      environment.systemPackages =
+        let
+          counter = pkgs.writers.writePython3Bin "counter" { } ''
+            from http.server import BaseHTTPRequestHandler, HTTPServer
+
+            counter = 0
+
+
+            class HTTPRequestHandler(BaseHTTPRequestHandler):
+                def do_POST(self):
+                    match self.path:
+                        case "/counter":
+                            global counter
+                            counter += 1
+                            self.send_response(204)
+                            self.end_headers()
+                        case _:
+                            self.send_response(404)
+                            self.end_headers()
+
+                def do_GET(self):
+                    match self.path:
+                        case "/counter":
+                            self.send_response(200)
+                            self.send_header("Content-type", "text/plain")
+                            self.end_headers()
+                            _ = self.wfile.write(str(counter).encode())
+                        case "/old":
+                            self.send_response(200)
+                            self.send_header("Content-type", "text/plain")
+                            self.end_headers()
+                            _ = self.wfile.write("fail".encode())
+                        case "/new":
+                            self.send_response(200)
+                            self.send_header("Content-type", "text/plain")
+                            self.end_headers()
+                            _ = self.wfile.write("success".encode())
+                        case _:
+                            self.send_response(404)
+                            self.end_headers()
+
+
+            server_address = ("", 8000)
+            server = HTTPServer(server_address, HTTPRequestHandler)
+            server.serve_forever()
+          '';
+        in
+        [
+          counter
+          pkgs.mitmproxy
+        ];
+    };
+
+  testScript =
+    let
+      addonScript = pkgs.writeText "addon-script" ''
+        def request(flow):
+            flow.request.path = "/new"
+      '';
+    in
+    ''
+      def curl(command: str, proxy: bool):
+          if proxy:
+              command = "curl --proxy 127.0.0.1:8080 --cacert ~/.mitmproxy/mitmproxy-ca-cert.pem " + command
+          else:
+              command = "curl " + command
+          return machine.succeed(command)
+
+      start_all()
+      machine.wait_for_unit("default.target")
+
+      # https://docs.mitmproxy.org/stable/concepts/certificates/#using-a-custom-certificate-authority
+      machine.succeed("mkdir -p ~/.mitmproxy")
+      machine.succeed("ln -s ${caCert} ~/.mitmproxy/mitmproxy-ca.pem")
+
+      machine.succeed("counter >/dev/null &")
+      machine.wait_for_open_port(8000)
+
+      # rewrite
```

Comment:
Maybe link to the docs here and in the `replay` section, if any interesting page exists? like:

- https://docs.mitmproxy.org/stable/mitmproxytutorial-modifyrequests/
- https://docs.mitmproxy.org/stable/mitmproxytutorial-replayrequests/